enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/LdapRealm.java

```diff
@@ -143,38 +144,45 @@ protected AuthenticationInfo queryForAuthenticationInfo( AuthenticationToken tok
     {
         if ( authenticationEnabled )
         {
-            String serverString = server( (JndiLdapContextFactory) ldapContextFactory );
-            try
+            if ( useSAMAccountName )
             {
-                AuthenticationInfo info =
-                        useStartTls ? queryForAuthenticationInfoUsingStartTls( token, ldapContextFactory ) :
-                        super.queryForAuthenticationInfo( token, ldapContextFactory );
-                securityLog.debug( withRealm( "Authenticated user '%s' against %s",
-                        token.getPrincipal(), serverString ) );
-                return info;
+                return queryForAuthenticationInfoSAM( token, ldapContextFactory );
             }
-            catch ( Exception e )
+            else
```

Comment:
The diff in this method looks big, but I interpret it as first checking if we can do the SAMAccountName login, and otherwise doing the same as before?